### PR TITLE
[modules] New magneto and pitot

### PR DESCRIPTION
**Pull request by fvantienen**
This adds the functionality for a new type of Magneto + pitot. The module transmits PPRZ messages which contain magneto and airspeed information.

### CODE REVIEW

File: conf/modules/mag_pitot_uart.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="mag_pitot_uart" dir="sensors">
+  <doc>
+    <description>Remotely located magnetometer and pitot tube over uart (RS232) communication</description>
```
Comment (flixr) [inline conf/modules/mag_pitot_uart.xml]: missing description of the configure options...

File: conf/modules/mag_pitot_uart.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <header>
+    <file name="mag_pitot_uart.h"/>
+  </header>
+  <init fun="mag_pitot_init()"/>
+  <event fun="mag_pitot_event()"/>
+  <makefile>
+    <configure name="MAG_PITOT_PORT" default="uart5" case="upper|lower"/>
+    <define name="USE_$(MAG_PITOT_PORT_UPPER)"/>
+    <define name="USE_$(MAG_PITOT_PORT_UPPER)_TX" value="FALSE"/>
+    <define name="$(MAG_PITOT_PORT_UPPER)_BAUD" value="$(MAG_PITOT_BAUD)"/>
```
Comment (flixr) [inline conf/modules/mag_pitot_uart.xml]: missing configure default for baud rate

File: sw/airborne/modules/sensors/mag_pitot_uart.c [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  /* Parse the mag-pitot message */
+  uint8_t msg_id = mp_msg_buf[1];
+  switch (msg_id) {
+
+  /* Got a magneto message */
+  case DL_IMCU_REMOTE_MAG: {
+    struct Int32Vect3 raw_mag;
+
+    // Read the raw magneto information
+    raw_mag.y = - DL_IMCU_REMOTE_MAG_mag_x(mp_msg_buf);
```
Comment (flixr) [inline sw/airborne/modules/sensors/mag_pitot_uart.c]: hardcoded sign conventions??
Isn't that what imu_to_mag would be for?

File: sw/airborne/modules/sensors/mag_pitot_uart.c [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+#endif
+
+/* Initialize the magneto and pitot */
+void mag_pitot_init() {
+  // Initialize transport protocol
+  pprz_transport_init(&mag_pitot.transport);
+
+  // Set the Imu to Magneto rotation
+  struct FloatEulers imu_to_mag_eulers =
+    {IMU_TO_MAG_PHI, IMU_TO_MAG_THETA, IMU_TO_MAG_PSI};
```
Comment (flixr) [inline sw/airborne/modules/sensors/mag_pitot_uart.c]: IMU_TO_MAG missing in the documentation

**Comment by fvantienen** [comment]
Fixed the comments ;)
One note on the rotation is that it cannot change the axis system of the magneto, but found a `MAG_*_SIGN` define which can solve this.

**Comment by flixr** [comment]
IMHO it makes much more sense to ensure a correct right handed magneto coordinate system on the sender side.
But setting mag signs (which is applied when scaling) also works...

**Comment by flixr** [comment]
Do you guys have the hw/sw for this external module available as well?
Plan to add it in paparazzi? Or maybe link to where the hw/sw for this external module can be found....

**Comment by fvantienen** [comment]
I think this could be released as soon as we tested it fully. Currently only done some static testing.
We made both the HW and SW, and it is pretty trivial so @dewagter will prolly release both.

Indeed the axis system could better be fixed on the sensor side, but for now I keep it like this and as soon as we update the firmware we will update the airframe file 👍

**Comment by dewagter** [comment]
Sure: you can have the external magneto sensor hardware and firmware (using
ppzlink on an atmega).
Should I put it on the hardware git?

**Comment by flixr** [comment]
That would be great.

**Pull request merged**